compliance/utils/services/github.py

```diff
@@ -413,6 +414,68 @@ def get_commit_details(self, repo, since, branch='master', path=None):
             opts['path'] = path
         return self._make_request('get', f'repos/{repo}/commits', params=opts)
 
+    def get_pull_requests(self, repo, since=None, **kwargs):
+        """
+        Retrieve a repository's pull request information.
+
+        :param repo: the organization/repository as a string.
+        :param since: the starting date/time of a pull request as a datetime.
+
+        :returns: the repo branch's pull request information
+        """
+        api_url = '/'.join(['repos', repo, 'pulls'])
+        pull_requests = []
+
+        self.session.headers.update(
+            {'Accept': 'application/vnd.github.v3+json'}
+        )
+
+        if since is None:
+            pull_requests = self._paginate_api(api_url, **kwargs)
+        else:
+            # sort in github API should be set to created.
+            sort = kwargs.get('sort', None)
+            if sort is None or sort != 'created':
+                kwargs['sort'] = 'created'
+
+            # Retrieve pull request information and
+            # filter the information by created_at field
+            params = kwargs
+            params['page'] = params.get('page', 1)
+            response = self._make_request(
+                'get', api_url, parse=False, params=params
+            )
+
+            max_page = 1
+            if 'Link' in response.headers:
+                # Link is only present if there are multiple pages
+                link = response.headers['Link']
+                urls = link.replace('>', '').replace('<', '').split()
+                parsed_url = urlparse(urls[2].strip(';'))
+                max_page = int(parse_qs(parsed_url.query)['page'][0])
+
+            while response:
+                for pull_request in response.json():
+                    created_at = pull_request['created_at']
+                    created_at = datetime.datetime.strptime(
+                        created_at, '%Y-%m-%dT%H:%M:%SZ'
+                    )
+
+                    if created_at < since:
+                        break
+
+                    pull_requests.append(pull_request)
+
+                params['page'] += 1
+                if params['page'] > max_page:
+                    response = False
+                else:
+                    response = self._make_request(
+                        'get', api_url, parse=False, params=params
+                    )
+
```

Comment:
I think there are a few things amiss here:

- It's probably not a good idea to manipulate the `kwargs` of the method because that would change the caller's kwargs content.  I see that similar is being done elsewhere in the class but that doesn't make it right.  We should fix that as well but that's outside of the scope of this PR.
- assigning `params` to `kwargs` essentially does the same as the previous bullet.
- I think your `sort` option processing is overcomplicated and should just be set to `created`.
- You also must ensure that the `direction` option is set to `desc`.
- IIUC max_page is the total number of pages needed to paginate through ALL of the pull requests.  Therefore you need to ensure that the `while` processing is stopped once the `created_at` dttm is found to be less than the `since` dttm regardless of the current page as it relates to the max_page.
- Also IMO, setting `response` to None rather than False seems better to me but that's just a personal preference.  I see that you took that logic from elsewhere in the class but again, I think in this case `None` is better that `False`.

Suggestions to remedy the above:

```suggestion
        self.session.headers.update(
            {'Accept': 'application/vnd.github.v3+json'}
        )
        if not since:
            return self._paginate_api(api_url, **kwargs)
        pull_requests = []
        params = {**kwargs}
        params['page'] = kwargs.get('page', 1)
        # Sort results by "created" in descending order
        params['sort'] = 'created'
        params['direction'] = 'desc'
        response = self._make_request(
            'get', api_url, parse=False, params=params
        )
        max_page = 1
        if 'Link' in response.headers:
            # Link is only present if there are multiple pages
            link = response.headers['Link']
            urls = link.replace('>', '').replace('<', '').split()
            parsed_url = urlparse(urls[2].strip(';'))
            max_page = int(parse_qs(parsed_url.query)['page'][0])
        while response:
            for pull_request in response.json():
                created_at = datetime.strptime(
                    pull_request['created_at'], '%Y-%m-%dT%H:%M:%SZ'
                )
                # Filter based on provided since datetime
                if created_at < since:
                    response = None
                    break
                pull_requests.append(pull_request)
            params['page'] += 1
            if params['page'] > max_page:
                response = None
            else:
                response = self._make_request(
                    'get', api_url, parse=False, params=params
                )
```